core/modules/layout/layout.theme.inc

```diff
@@ -127,9 +127,17 @@ function theme_layout_template_option($variables) {
 function theme_layout_template_info($variables) {
   $template_info = $variables['template_info'];
 
-  // Build template thumbnail preview
-  $preview_path = $template_info['path'] . '/' . $template_info['preview'];
-  $preview_img = theme('image', array('path' => $preview_path));
+  // Build template thumbnail preview.
+  if (!empty($template_info['flexible'])) {
```

Comment:
Rather than checking if it's a flexible template, perhaps use `file_valid_uri($template_info['preview'])` to allow using a URI on all templates?